src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -97,7 +97,8 @@
     private final String[] argNames;                 // Generated names for the constructor arguments
     private final ClassDesc[] argDescs;              // Type descriptors for the constructor arguments
     private final String lambdaClassName;            // Generated name for the generated class "X$$Lambda$1"
-    private final ClassDesc lambdaClassDesc;         // Type descriptor for the generated class "X$$Lambda$1"
+    private final ConstantPoolBuilder pool = ConstantPoolBuilder.of();
+    private final ClassEntry lambdaClassEntry;       // Class entry for the generated class "X$$Lambda$1"
```

Comment:
Does this improvement go away if remove this change to use explicit class entries? I think I can do some constant pool work to improve ClassEntry retrieval from ClassDesc. Something like #20667, but I will try move the hash to ClassEntry to enforce ClassDesc primacy.

Reply:
When experimenting with this all the changes in this PR individually moved the needle in the right direction. If you have doubts about this facet in particular I can split it out and we can re-evaluate it together with #20667.